library/src/main/java/com/novoda/downloadmanager/lib/DownloadManagerModules.java

```diff
@@ -0,0 +1,101 @@
+package com.novoda.downloadmanager.lib;
+
+import android.content.Context;
+
+import com.novoda.downloadmanager.notifications.CancelledNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.CompleteNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.DownloadingNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.FailedNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.NotificationImageRetriever;
+import com.novoda.downloadmanager.notifications.QueuedNotificationCustomiser;
+
+public interface DownloadManagerModules {
+
+    NotificationImageRetriever getNotificationImageRetriever();
+
+    DownloadClientReadyChecker getDownloadClientReadyChecker();
+
+    QueuedNotificationCustomiser getQueuedNotificationCustomiser();
+
+    DownloadingNotificationCustomiser getDownloadingNotificationCustomiser();
+
+    CompleteNotificationCustomiser getCompleteNotificationCustomiser();
+
+    CancelledNotificationCustomiser getCancelledNotificationCustomiser();
+
+    FailedNotificationCustomiser getFailedNotificationCustomiser();
+
+    class Builder {
+
+        private final Context context;
+
+        private QueuedNotificationCustomiser queuedCustomiser;
```

Comment:
if someone passes null we could throw a dev error or annotate the `withBlaBla` with `@notnull`